xrpl/wallet/main.py

```diff
@@ -1,44 +1,43 @@
 """The information needed to control an XRPL account."""
-
 from __future__ import annotations
 
-from typing import Type
+from typing import Optional, Type
 
-from xrpl import CryptoAlgorithm
 from xrpl.core.keypairs import derive_classic_address, derive_keypair, generate_seed
 
 
 class Wallet:
     """The information needed to control an XRPL account."""
 
-    def __init__(self: Wallet, seed: str) -> None:
+    def __init__(self: Wallet, pub_key: str, priv_key: str) -> None:
         """
         Generate a new Wallet.
 
         Args:
-            seed: The seed from which the public and private keys are derived.
+            pub_key: The public key for this Wallet.
+            priv_key: The private key for this Wallet.
+
         """
-        self.seed = seed
-        self.pub_key, self.priv_key = derive_keypair(self.seed)
+        self.pub_key = pub_key
+        self.priv_key = priv_key
         self.classic_address = derive_classic_address(self.pub_key)
         self.next_sequence_num = 0
```

Comment:
how are we going to handle this if users are loading an existing account? what happens in ripple-lib?

Reply:
ripple-lib has:

- [generateXaddress](https://xrpl.org/rippleapi-reference.html#generatexaddress) - optionally takes explicitly specified entropy, doesn't let you specify the seed.
- [generateAddress (deprecated, nearly the same anyway)](https://xrpl.org/rippleapi-reference.html#generateaddress)
- [isValidSecret](https://xrpl.org/rippleapi-reference.html#isvalidsecret) to check a seed
- [deriveKeypair](https://xrpl.org/rippleapi-reference.html#derivekeypair) from seed
- [deriveAddress](https://xrpl.org/rippleapi-reference.html#deriveaddress) from pubkey

rippled's wallet_propose accepts the kitchen sink in seed formats, but weirdly specifying an actual private key isn't an option, and it doesn't expose any methods for going from private keys to public keys or from public keys to addresses.

Reply:
right right i get all that, but i mean let's say you have an account and its keys or seed. you've already made some transactions on it. now you want to wrap those keys in a `Wallet` object so you can use it with this lib. how do we correctly initialize the `next_sequence_num`, since in this case it is not `0`?

Reply:
Ah, I think I see what you mean. IMO the user should call some account info lookup function and then set the next_sequence_num attribute directly.

Is it too late to rename that field to just `sequence`?

Reply:
Could we initialize it to `None` instead, so it'll error if you try to use it on your own?

Reply:
I was serious about renaming it to `sequence`, by the way. I would even be in favor of `seq` since people seem to make typos on the word "sequence" surprisingly often, and it's still pretty unambiguous.

Yeah, it's a breaking change, but if it significantly improves the developer experience (which I think it does) then better to do that now rather than changing it later when it breaks more people's code.

Reply:
idea: add a required field to `__init__` of `seq_num`. then `create` will pass `0` to that explicitly whereas user will know they need to determine it

Reply:
Can we do this in the wallet-cleanup PR (I'll work on it today)? I'd like to get the rest of these changes pushed.